core/src/apps/homescreen/homescreen.py

```diff
@@ -33,3 +53,39 @@ def on_render(self) -> None:
         # homescreen with shifted avatar and text on bottom
         ui.display.avatar(48, 48 - 10, self.image, ui.WHITE, ui.BLACK)
         ui.display.text_center(ui.WIDTH // 2, 220, self.label, ui.BOLD, ui.FG, ui.BG)
+
+        self.repaint = False
+
+    def on_touch_start(self, _x: int, _y: int) -> None:
+        if self.loader.start_ms is not None:
+            self.loader.start()
+        elif config.has_pin():
+            self.touch_ms = utime.ticks_ms()
+
+    def on_touch_end(self, _x: int, _y: int) -> None:
+        if self.loader.start_ms is not None:
+            self.repaint = True
+        self.loader.stop()
+        self.touch_ms = None
+
+        # raise here instead of self.loader.on_finish so as not to send TOUCH_END to the lockscreen
+        if self.loader.elapsed_ms() >= self.loader.target_ms:
+            raise ui.Result(None)
+
+    def _loader_start(self) -> None:
+        ui.display.clear()
+        ui.display.text_center(ui.WIDTH // 2, 35, "Hold to lock", ui.BOLD, ui.FG, ui.BG)
+        self.loader.start(_LOADER_DELAY_MS)
```

Comment:
One more thing, I'm not too convinced about this. It feels jarring that when the loader is shown, it's already a bit up.

Reply:
Removed in 95bbc9e.